Share with Gifski/ShareViewController.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  ShareViewController.swift
+//  Share with Gifski
+//
+//  Created by Koray Koska on 12.09.19.
+//  Copyright © 2019 Sindre Sorhus. All rights reserved.
+//
+
+import Cocoa
+import WebKit
+
+class ShareViewController: NSViewController {
+
+    override var nibName: NSNib.Name? {
+        return NSNib.Name("ShareViewController")
+    }
+
+    override func loadView() {
+		super.loadView()
+
+		guard let item = (self.extensionContext?.inputItems[0] as? NSExtensionItem)?.attachments?.first else {
+			errorOpenMainApp()
+			return
+		}
+
+		var typeIdentifier: String
+		if item.hasItemConformingToTypeIdentifier("public.mpeg-4") {
+			typeIdentifier = "public.mpeg-4"
+		} else if item.hasItemConformingToTypeIdentifier("com.apple.m4v-video") {
+			typeIdentifier = "com.apple.m4v-video"
+		} else if item.hasItemConformingToTypeIdentifier("com.apple.quicktime-movie") {
+			typeIdentifier = "com.apple.quicktime-movie"
+		} else {
+			errorOpenMainApp()
+			return
+		}
+
+		item.loadFileRepresentation(forTypeIdentifier: typeIdentifier) { url, _ in
+			guard let url = url else {
+				self.errorOpenMainApp()
```

Comment:
Same here. More useful message.

Side note: Are you sure the share extension cannot present the error message itself?

Reply:
It seems better to me to handle all errors in one place and not some in the main app and some in the extension. I will pass the error message though.

Reply:
Why is it better? Generally, it's better to show an error as close to the context/source as possible. If it is possible to show an error alert in the share extension, I would prefer that.

Reply:
I don't see how actually. I tried it initially and now again. We don't have access to a window in an extension as it runs in the context of another app and `runModal()` crashes.
Maybe some kind of custom view embedded into the alert view which shows the message and an OK button?